codecs-parent/codecs-stream/src/test/java/org/interledger/codecs/stream/StreamPacketFixturesTest.java

```diff
@@ -83,6 +93,89 @@ public StreamPacketFixturesTest(StreamTestFixture streamTestFixture) {
+  private static boolean checkFixtureRFCStalenessState() {
+    Properties properties = readProperties();
+    String fixtureUrl = (String) properties.get("stream.packetFixtures.file");
+    String expectedCheckSum = (String) properties.get("stream.packetFixtures.checksum");
+    String fileName = (String) properties.get("stream.packetFixtures.fileName");
+
+    byte[] downloadedFixtureData;
+    downloadedFixtureData = downloadFixtureFile(fixtureUrl, fileName);
+
+    Objects.requireNonNull(downloadedFixtureData);
+    if (downloadedFixtureData.length > 0) {
+      String computedCheckSum = BaseEncoding.base16().encode(checksum(downloadedFixtureData)).toLowerCase();
+
+      // File matches the expectation therefore the fixtures haven't changed in the RFCs. We can proceed with
+      // running the test of the fixtures.
+      return computedCheckSum.equals(expectedCheckSum);
+    }
+    return false;
+  }
+
+  /**
+   * When the test is run, this method reads the local fixtures resource and performs a checksum on the file contents
+   * to ensure the integrity of the file and compares it with the expected {@code stream.packetFixtures.checksum} value
+   * in the properties configuration located at {@code fixtures.properties}.
+   *
+   * <p>This test is present to ensure that in case the file is updated locally, we also update the corresponding
+   * checksum of the file.</p>
+   *
+   * @return {@code false} if the fixtures have changed locally, else {@code true}.
+   */
+  private static boolean checkLocalFixtureFileState() {
+    Properties properties = readProperties();
+    String expectedCheckSum = (String) properties.get("stream.packetFixtures.checksum");
+    String fileName = (String) properties.get("stream.packetFixtures.fileName");
+
+    try {
+      Path path = Paths.get(StreamPacketFixturesTest.class.getClassLoader().getResource(fileName).toURI());
+      byte[] buffer = Files.readAllBytes(path);
+      Objects.requireNonNull(buffer);
+      String computedCheckSum = BaseEncoding.base16().encode(checksum(buffer)).toLowerCase();
+      return computedCheckSum.equals(expectedCheckSum);
+    } catch (URISyntaxException | IOException e) {
+      e.printStackTrace();
+    }
+
+    return false;
+  }
+
+  /**
+   * The {@link ClassRule} provides a {@code before()} method which is executed before the tests in the class
+   * are executed. As a part of the class rule, we expect the validation of the checksum to pass on both the local
+   * test fixtures file and the remote test fixtures file in the RFCs repository. Any change to either the local or
+   * the remote file will result in a failure of this test.
+   */
+  @ClassRule
+  public static ExternalResource externalResource = new ExternalResource() {
+    @Override
+    protected void before() throws Throwable {
+      boolean rfcStatus = checkFixtureRFCStalenessState();
+      boolean localFixtureStatus = checkLocalFixtureFileState();
```

Comment:
Overall, I like the design/intention of this PR. Only thing is that the test should pass if the system running the test is offline or otherwise unable to download the remote fixture. So, what if we just return `false` from `checkFixtureRFCStalenessState()` if it can't download?

Reply:
This has been fixed.